OpenRA.Mods.Common/Activities/DeliverResources.cs

```diff
@@ -57,7 +62,10 @@ public override Activity Tick(Actor self)
 
 			// No refineries exist; check again after delay defined in Harvester.
 			if (harv.LinkedProc == null)
-				return ActivityUtils.SequenceActivities(new Wait(harv.Info.SearchForDeliveryBuildingDelay), this);
+			{
+				ChildActivity = new Wait(harv.Info.SearchForDeliveryBuildingDelay);
```

Comment:
I believe these should be using `QueueChild` rather than assigning `ChildActivity` directly.

Reply:
Fixed.